ui-tests/src/test/java/org/openmrs/reference/VisitNoteTest.java

```diff
@@ -51,25 +49,31 @@ public void VisitNoteTest() throws Exception {
         visitNotePage.addSecondaryDiagnosis(DIAGNOSIS_SECONDARY);
         visitNotePage.addNote("This is a note");
         patientDashboardPage = visitNotePage.save();
-        assertEquals(DIAGNOSIS_PRIMARY, visitNotePage.primaryDiagnosis());
-        assertEquals(DIAGNOSIS_SECONDARY, visitNotePage.secondaryDiagnosis());
-
-        PatientVisitsDashboardPage patientVisitsDashboardPage = patientDashboardPage.goToRecentVisits();
-
+        patientDashboardPage.waitForPage();
+        patientVisitsDashboardPage = patientDashboardPage.goToRecentVisits();
         EditVisitNotePage editVisitNotePage = patientVisitsDashboardPage.goToEditVisitNote();
-
-        editVisitNotePage.deleteDiagnosis();
+        
+        //Editing visit Note 
+        editVisitNotePage.waitForPage();
+        editVisitNotePage.removeDiagnosis();
+        editVisitNotePage.clearNote();
+        editVisitNotePage.addNote("This is edited note");
         editVisitNotePage.addSecondaryDiagnosis(DIAGNOSIS_SECONDARY_UPDATED);
-
-        //TODO Edit function doesn't work int02
-        patientDashboardPage = editVisitNotePage.save();
         assertEquals(DIAGNOSIS_SECONDARY_UPDATED, patientDashboardPage.secondaryDiagnosis());
-        patientVisitsDashboardPage = patientDashboardPage.goToRecentVisits();
-
-        //TODO Delete function doesn't work on int02
-        patientVisitsDashboardPage.deleteVisitNote();
-        patientDashboardPage.confirmDeletion();
-        //Assert that visit note is not present on encounter list and its done
+        patientDashboardPage = visitNotePage.save();
+      
+        //Add a new  diagnosis
+        patientDashboardPage.goToVisitNote();
+        editVisitNotePage.selectProviderAndLocation();
+        editVisitNotePage.addDiagnosis(DIAGNOSIS_PRIMARY);
+        editVisitNotePage.addSecondaryDiagnosis(DIAGNOSIS_SECONDARY);
+        editVisitNotePage.addNote("This is a new note");
+        patientDashboardPage = visitNotePage.save();
+        
+        //Deleting visit Note.....
+        visitNotePage.viewVisitNote();
+        visitNotePage.deleteDiagnosis();
+        visitNotePage.confirmDeleteDiagnosis();
```

Comment:
Should this be done here or in the teardown? If here, should we split this into another test?

Reply:
Thanks @ibacher  , if we split them we it would look somewhat not straight especially maintaining them , so we might need to make them in one single test for easy workflow